File: python/1431. Kids With the Greatest Number of Candies.py
class Solution:
    def kidsWithCandies(self, candies: List[int], extraCandies: int) -> List[bool]:
        output = []
        max = candies[0]
        i =1
        while (i <len(candies)):
            if (candies[i]> max):
                max=candies[i]
            i+=1
        for c in candies:
            if ((extraCandies + c) >= max ):
                output.append(True)
            else:
                output.append(False)
        return output

File: javascript/1431. Kids With the Greatest Number of Candies.js
/**
 * @param {number[]} candies
 * @param {number} extraCandies
 * @return {boolean[]}
 */
var kidsWithCandies = function(candies, extraCandies) {
    let max = candies[0];
    let i = 0;
    while (i < candies.length)
    {
        if (candies[i] > max)
            max = candies[i];
        i++;
    }
    const output = [];
    i=0;
    while (i< candies.length)
    {
        if ((candies[i]+extraCandies) >= max)
             output.push(true);
        else 
            output.push(false);
        i++;
    }
    return output;
};